rightscale/rsc/client_test.go

```diff
@@ -290,3 +290,47 @@ func TestOnlyPopulated(t *testing.T) {
 		t.Errorf("Result of onlyPopulated was incorrect, got: %v, expected: %v", testFields.onlyPopulated(), expectedResult)
 	}
 }
+
+func TestAnalyzeSource(t *testing.T) {
+	const (
+		invalidSource = `
+define main( do
+	foo
+end
+`
+		goodSourceNoReturn = `
+define main() do
+	bar
+end
+`
+		goodSourceWithReturn = `
+define main() return $out1, $out2 $suma do
+	$out1 = 156.5534
+	$out2 = 42421000
+	$suma = $out1 + $out2
+end
+`
+	)
+
+	var sourcetests = []struct {
+		source         string
+		valid          bool
+		expectsOutputs bool
+	}{
+		{invalidSource, false, false},
+		{goodSourceNoReturn, true, false},
+		{goodSourceWithReturn, true, true},
+	}
+
+	for _, tt := range sourcetests {
+		t.Run(tt.source, func(t *testing.T) {
```

Comment:
You probably want to give each test a name rather than use the source as name (add a initial field in the struct `name string`).